src/experiments.py

```diff
@@ -0,0 +1,572 @@
+            getattr(cls, key) >= from_datetime,
+            getattr(cls, key) <= to_datetime
+        )
+    elif from_datetime is not None:
+        constructed_filter[key] = (
+            getattr(cls, key) >= from_datetime
+        )
+    elif to_datetime is not None:
+        constructed_filter[key] = (
+            getattr(cls, key) <= to_datetime
+        )
+
+    print(f'constructed_filter: {constructed_filter}')
+    return constructed_filter
+
+
+def get_string_filter(filters, cls, key, constructed_filter):
+    if not isinstance(filters, dict):
+        msg = f'Invalid type for filters, must be \'dict\', actually ' \
+            f'type: {type(filters)}'
+        raise TypeError(msg)
+
+    print(f'Column \'{key}\' is of type {type(getattr(cls, key).type)}.')
+    string_flt = filters.get(key)
+
+    if string_flt is None:
+        print(f'No \'{key}\' filter detected')
+        return constructed_filter
+
+    like_filter = string_flt.get('like')
+    # prefer like search over exact match if both exist
+    if like_filter is not None:
+        constructed_filter[key] = (getattr(cls, key).like(like_filter))
+        return constructed_filter
+
+    exact_match_filter = string_flt.get('exact')
+    if exact_match_filter is not None:
+        constructed_filter[key] = (getattr(cls, key) == exact_match_filter)
+
+    return constructed_filter
+
+
+def construct_filters(filters):
+    constructed_filter = {}
+
+    constructed_filter = get_string_filter(
+        filters, exp, 'name', constructed_filter)
+
+    constructed_filter = get_time_filter(
+        filters, exp, 'cycle_start', constructed_filter)
+
+    constructed_filter = get_time_filter(
+        filters, exp, 'cycle_stop', constructed_filter)
+
+    constructed_filter = get_time_filter(
+        filters, exp, 'wallclock_start', constructed_filter)
+
+    constructed_filter = get_time_filter(
+        filters, exp, 'wallclock_end', constructed_filter)
+
+    constructed_filter = get_string_filter(
+        filters, exp, 'owner_id', constructed_filter)
+
+    constructed_filter = get_string_filter(
+        filters, exp, 'group_id', constructed_filter)
+    
+    return constructed_filter
+
+
+def validate_column_name(cls, value):
+    if not isinstance(value, str):
+        raise TypeError(f'Column name must be a str, was {type(value)}')
+
+    try:
+        column_obj = getattr(cls, value)
+        print(f'column: {column_obj}, type(key): {type(column_obj)}')
+    except Exception as err:
+        msg = f'Column does not exist - err: {err}'
+        raise ValueError(msg)
+    
+    return column_obj
+
+
+def validate_order_dir(value):
+    if not isinstance(value, str):
+        raise TypeError(f'\'order_by\' must be a str, was {type(value)}')
+    
+    if value not in VALID_ORDER_BY:
+        raise TypeError(f'\'order_by\' must be one of {VALID_ORDER_BY}, ' \
+            f' was {value}')
+    
+    return value
+
+
+def build_column_ordering(cls, ordering):
```

Comment:
add comment that this relates to SQL ORDER BY

Reply:
added method comment section